lib/inspec/waiver_file_reader.rb

```diff
@@ -15,34 +17,50 @@ def self.read_waivers_from_file(profile_id, files)
       output = {}
 
       files.each do |file_path|
-        file_extension = File.extname(file_path)
-        data = nil
-        if [".yaml", ".yml"].include? file_extension
-          data = Secrets::YAML.resolve(file_path)
-          unless data.nil?
-            data = data.inputs
-            validate_json_yaml(data)
-          end
-        elsif file_extension == ".csv"
-          data = Waivers::CSVFileReader.resolve(file_path)
-          headers = Waivers::CSVFileReader.headers
-          validate_headers(headers)
-        elsif file_extension == ".json"
-          data = Waivers::JSONFileReader.resolve(file_path)
-          validate_json_yaml(data) unless data.nil?
-        end
-        output.merge!(data) if !data.nil? && data.is_a?(Hash)
-
-        if data.nil?
-          raise Inspec::Exceptions::WaiversFileNotReadable,
-              "Cannot find parser for waivers file '#{file_path}'. " \
-              "Check to make sure file has the appropriate extension."
-        end
+        next unless valid_waiver_file?(file_path)
+
+        data = parse_waiver_file(file_path)
+        output.merge!(data) if data.is_a?(Hash)
       end
 
       @waivers_data[profile_id] = output
     end
 
+    def self.valid_waiver_file?(file_path)
+      # Check if the file is readable
+      file_extension = File.extname(file_path)
+      unless SUPPORTED_FILE_EXTENSION.include?(file_extension)
```

Comment:
functionally okay. But when the file doesn't exist this condition will raise UnsupportedFileExtension while we should raise as file doesn't exist.